pages/api/search/[[...slug]].js

```diff
@@ -0,0 +1,59 @@
+import { getUsers } from "../profiles";
+import logger from "@config/logger";
+
+export default async function handler(req, res) {
+  const { slug } = req.query;
+
+  if (req.method != "GET") {
+    return res
+      .status(400)
+      .json({ error: "Invalid request: GET request required" });
+  }
+
+  if (!slug) {
+    return res
+      .status(400)
+      .json({ error: "Invalid request: search input is required" });
+  }
+
+  const cleanedSlug = slug
+    .join()
+    .trim()
+    .replace(/\s{2,}/g, " ")
+    .toLowerCase();
+  const terms = cleanedSlug.split(",");
+  let filteredUsers = [];
+
+  try {
+    const users = await getUsers();
+
+    filteredUsers = users.flat().filter((user) => {
+      const nameLower = user.name.toLowerCase();
+      const usernameLower = user.username.toLowerCase();
+      const userTagsString = user.tags.join(", ").toLowerCase();
+      const userLocationString = user.location
+        ? user.location.provided.toLowerCase()
+        : "";
+
+      const isUserMatched = terms.every((term) => {
+        const cleanedTerm = term.trim();
+
+        return (
+          usernameLower.includes(cleanedTerm) ||
+          nameLower.includes(cleanedTerm) ||
+          userTagsString.includes(cleanedTerm) ||
+          userLocationString.includes(cleanedTerm)
+        );
+      });
+      return isUserMatched;
+    });
```

Comment:
Could this be reduced by having Mongo perform the search? This could also come in another PR as I don't want to add more complexity to this PR

Reply:
Yes here we need mongo 👍 but currently I don't have much knowledge in mongo so i performed the search logic by javascript